### PR TITLE
Finish up GPA layer cleanup with unique_objects

### PR DESCRIPTION
**Pull request by mark-lunarg**
All done but for swapchain, which is not long for this life.

### CODE REVIEW

File: layers/unique_objects.cpp [hunk 1]
```diff
@@ -268,46 +248,31 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
...
     }
 
-    layer_data *dev_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
-    VkLayerDispatchTable *disp_table = &dev_data->dispatch_table;
-    if (disp_table->GetDeviceProcAddr == NULL) {
-        return NULL;
-    }
-    return disp_table->GetDeviceProcAddr(device, funcName);
+    layer_data *device_data = GetLayerDataPtr(get_dispatch_key(device), layer_data_map);
+    auto &table = device_data->dispatch_table;
```
Comment (tobine) [inline layers/unique_objects.cpp]: I believe const preferred here.

File: layers/unique_objects.cpp [hunk 2]
```diff
@@ -268,46 +248,31 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
...
-    if (addr) {
-        return addr;
-    }
-
-    VkLayerInstanceDispatchTable *disp_table = &instance_data->dispatch_table;
-    if (disp_table->GetInstanceProcAddr == NULL) {
-        return NULL;
-    }
-    return disp_table->GetInstanceProcAddr(instance, funcName);
+    auto &table = instance_data->dispatch_table;
```
Comment (tobine) [inline layers/unique_objects.cpp]: const

**Comment by chrisforbes** [comment]
:+1:  Yes, can have const there. Doesn't buy you a lot though.

**Comment by mark-lunarg** [comment]
Updated consts and pushed to Master.

**Pull request merged**